app/modules/topics/components/Editor/index.js

```diff
@@ -47,19 +66,58 @@ class PureEditor extends React.Component<Props> {
     onSave();
   };
 
+  beforeUnloadHandler = (event: Event): boolean => {
+    const { topic } = this.props;
+
+    if (topic.isDirty) {
+      // Cancel the event as stated by the standard
+      event.preventDefault();
+
+      // Chrome requires returnValue to be set
+      /* eslint-disable no-param-reassign */
+      // $FlowFixMe flowtype for Event does not contain the `returnValue` property
+      event.returnValue = '';
+      /* eslint-enable */
+    }
+
+    return topic.isDirty;
+  }
+
+  componentWillMount = (): void => {
```

Comment:
Note: `componentWillMount` [is deprecated](https://reactjs.org/docs/react-component.html#mounting). I think this will work just as well if you use `componentDidMount` instead?